pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -181,23 +147,96 @@ func LoadTracer(cfg *config.Config, mgrOpts manager.Options, perfHandlerTCP *dde
+
+	ringbufferEnabled := RingBufferSupported(config)
+	AddBoolConst(&mgrOpts, ringbufferEnabled, "ringbuffer_enabled")
+	tcpEnabled := config.CollectTCPv4Conns || config.CollectTCPv6Conns
+
+	if err := initManager(m, connCloseEventHandler, runtimeTracer, config); err != nil {
 		return nil, nil, fmt.Errorf("could not initialize manager: %w", err)
 	}
 
+	if ringbufferEnabled {
+		mgrOpts.MapSpecEditors[probes.ConnCloseEventMap] = manager.MapSpecEditor{
+			Type:       ebpf.RingBuf,
+			MaxEntries: uint32(ComputeDefaultClosedConnRingBufferSize()),
+			KeySize:    0,
+			ValueSize:  0,
+			EditorFlag: manager.EditType | manager.EditMaxEntries | manager.EditKeyValue,
+		}
+	}
+	protocolClassificationTailCalls := []manager.TailCallRoute{
+		{
+			ProgArrayName: probes.ClassificationProgsMap,
+			Key:           netebpf.ClassificationQueues,
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: probes.ProtocolClassifierQueuesSocketFilter,
+				UID:          probeUID,
+			},
+		},
+		{
+			ProgArrayName: probes.ClassificationProgsMap,
+			Key:           netebpf.ClassificationDBs,
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: probes.ProtocolClassifierDBsSocketFilter,
+				UID:          probeUID,
+			},
+		},
+		{
+			ProgArrayName: probes.ClassificationProgsMap,
+			Key:           netebpf.ClassificationGRPC,
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: probes.ProtocolClassifierGRPCSocketFilter,
+				UID:          probeUID,
+			},
+		},
+		{
+			ProgArrayName: probes.TCPCloseProgsMap,
+			Key:           0,
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: probes.TCPCloseFlushReturnPerfbuffer,
+				UID:          probeUID,
+			},
+		},
+	}
+	tailCallsIdentifiersSet := make(map[manager.ProbeIdentificationPair]struct{}, len(protocolClassificationTailCalls)+2)
+	if tcpEnabled {
+		probePair := manager.ProbeIdentificationPair{
+			EBPFFuncName: probes.TCPConnCloseEmitEventPerfbuffer,
+			UID:          probeUID,
+		}
+		if ringbufferEnabled {
+			probePair.EBPFFuncName = probes.TCPConnCloseEmitEventRingbuffer
+		}
+		mgrOpts.TailCallRouter = append(mgrOpts.TailCallRouter, []manager.TailCallRoute{
+			{
+				ProgArrayName:           probes.ConnCloseProgsIndvMap,
+				Key:                     0,
+				ProbeIdentificationPair: probePair,
+			},
+		}...)
+		tailCallsIdentifiersSet[probePair] = struct{}{}
+	}
+
 	var undefinedProbes []manager.ProbeIdentificationPair
 
 	var closeProtocolClassifierSocketFilterFn func()
 	classificationSupported := ClassificationSupported(config)
-	addBoolConst(&mgrOpts, classificationSupported, "protocol_classification_enabled")
+	AddBoolConst(&mgrOpts, classificationSupported, "protocol_classification_enabled")
 
 	if classificationSupported {
+		if ringbufferEnabled {
+			protocolClassificationTailCalls[3].ProbeIdentificationPair.EBPFFuncName = probes.TCPCloseFlushReturnRingbuffer
```

Comment:
this is error prune, as any change in the original list can break us.
instead you should use `slices.Find` or `slices.DeleteFunc`, to have a smart search / deletion, and after that, to replace the correct method

example for DeleteFunc - 
https://github.com/DataDog/datadog-agent/blob/294ae46153c39545b7e9517a65fed146f50827fb/pkg/ebpf/telemetry/perf_metrics.go#L172-L174

Reply:
thanks! removed this code by refactoring it a bit